hazelcast-sql/src/main/java/com/hazelcast/sql/impl/FieldsUtil.java

```diff
@@ -83,6 +88,22 @@ public static SortedMap<String, Class<?>> resolveClass(@Nonnull Class<?> clazz)
         return fields;
     }
 
+    /**
+     * Constructs instance-to-name mapping for enum-like classes.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> Map<T, String> getEnumConstants(Class<T> type) {
+        return Arrays.stream(QueryDataType.class.getDeclaredFields())
```

Comment:
```suggestion
        return Arrays.stream(type.getDeclaredFields())
```
you could add some unit tests

Reply:
Moved to the upcoming PR.